src/argilla/client/feedback/training/schemas.py

```diff
@@ -1064,13 +1130,208 @@ def _dict_to_format(ds: datasets.Dataset) -> List[Dict[str, List[Dict[str, str]]
+
+    @requires_version("sentence-transformers")
+    def _prepare_for_training_with_sentence_transformers(
+        self, data: List[dict], train_size: float, seed: int
+    ) -> Union["InputExample", Tuple["InputExample", "InputExample"]]:
+        from sentence_transformers import InputExample
+
+        # Use the first sample to decide what type of dataset to generate:
+        sample_keys = data[0].keys()
+        if "label" in sample_keys:
+            if (len(sample_keys) == 3) and all(s in sample_keys for s in ["sentence-1", "sentence-2"]):
+                dataset_fields = lambda sample: {
+                    "texts": [sample["sentence-1"], sample["sentence-2"]],
+                    "label": sample["label"],
+                }
+            elif all(s in sample_keys for s in ["sentence-1", "sentence-2", "sentence-3"]):
+                dataset_fields = lambda sample: {
+                    "texts": [sample["sentence-1"], sample["sentence-2"], sample["sentence-3"]],
+                    "label": sample["label"],
+                }
+            elif (len(sample_keys) == 2) and ("sentence" in sample_keys):
+                raise ValueError(
+                    "Datasets containing a `sentence` and a `label` should be transformed "
+                    "to contain triplets of `sentence-1`, `sentence-2`, `sentence-3` and `label`."
+                    r"An example can be seen at: https://github.com/UKPLab/sentence-transformers/blob/master/examples/training/other/training_batch_hard_trec.py"
+                )
+            else:
+                raise ValueError(
+                    "Labeled datasets must contain a pair of `sentence-1` and "
+                    "`sentence-2` or triplets `sentence-1`, `sentence-2`, `sentence-3` "
+                    "as well as a `label`."
+                )
+        else:
+            if len(sample_keys) == 2:
+                dataset_fields = lambda sample: {"texts": [sample["sentence-1"], sample["sentence-2"]]}
+            elif len(sample_keys) == 3:
+                dataset_fields = lambda sample: {
+                    "texts": [sample["sentence-1"], sample["sentence-2"], sample["sentence-3"]]
+                }
+            else:
+                raise ValueError(
+                    "Unlabeled datasets must contain a pair of `sentence-1` and "
+                    "`sentence-2` or triplets `sentence-1`, `sentence-2`, `sentence-3`."
+                )
+
+        train_samples = []
+        for sample in data:
+            if isinstance(sample, list):
+                for record in sample:
+                    train_samples.append(InputExample(**dataset_fields(record)))
+            else:
+                train_samples.append(InputExample(**dataset_fields(sample)))
+
+        if train_size != 1:
+            stratify = None
+            if label := train_samples[0].label:
+                if isinstance(label, int):
+                    stratify = [example.label for example in train_samples]
```

Comment:
Nice, I love this operator. I think we can only do these since 2 months ago or so.
That said, I think we can simplify this:
```suggestion
            if (label := train_samples[0].label) and isinstance(label, int):
                stratify = [example.label for example in train_samples]
```
I'm not 100% sure on this though.

Reply:
Also, I see now that you need the labels to be an int here, so the "Can you just use float?" comment from above is a bit moot

Reply:
I checked the 'resolve conversation' here but didn't commit the suggestion. It does work so your simplified version wins